src/System.Text.Encodings.Web/src/System/Text/Unicode/UnicodeRanges.cs

```diff
@@ -17,31 +17,31 @@ public static partial class UnicodeRanges
         /// An empty <see cref="UnicodeRange"/>. This range contains no code points.
         /// </summary>
         public static UnicodeRange None => _none ?? CreateEmptyRange(ref _none);
-        private static UnicodeRange _none;
+        private static UnicodeRange? _none;
 
         /// <summary>
         /// A <see cref="UnicodeRange"/> which contains all characters in the Unicode Basic
         /// Multilingual Plane (U+0000..U+FFFF).
         /// </summary>
         public static UnicodeRange All => _all ?? CreateRange(ref _all, '\u0000', '\uFFFF');
-        private static UnicodeRange _all;
+        private static UnicodeRange? _all;
 
         [MethodImpl(MethodImplOptions.NoInlining)] // the caller should be inlined, not this method
-        private static UnicodeRange CreateEmptyRange(ref UnicodeRange range)
+        private static UnicodeRange CreateEmptyRange(ref UnicodeRange? range)
         {
             // If the range hasn't been created, create it now.
```

Comment:
Nit: not related to your change, but this comment is strange.  This method is going to create and store the range regardless of whether it's already been created.